vllm/transformers_utils/tokenizers/mistral.py

```diff
@@ -164,7 +164,19 @@ def make_mistral_chat_completion_request(
         if message.get("role") in ("assistant", "tool"):
             content = message.get("content")
             if isinstance(content, list):
-                content = "\n".join(chunk.get("text") for chunk in content)
+                aggregated_content: list[dict[str, Any]] = []
+                for chunk in content:
+                    if chunk.get(
+                            "type"
+                    ) == "text" and aggregated_content and aggregated_content[
+                            -1].get("type") == "text":
+                        aggregated_content[-1]["text"] += "\n\n" + chunk.get(
+                            "text")
+                    else:
+                        aggregated_content.append(chunk)
+                if len(aggregated_content) == 1 and aggregated_content[0].get(
+                        "type") == "text":
+                    content = aggregated_content[0]["text"]
```

Comment:
```suggestion
                aggregated_content: list[dict[str, Any]] = []
                for chunk in content:
                    if chunk.get(
                            "type"
                    ) == "text" and aggregated_content and aggregated_content[
                            -1].get("type") == "text":
                        aggregated_content[-1]["text"] += "\n\n" + chunk.get(
                            "text")
                    else:
                        aggregated_content.append(chunk)
                if len(aggregated_content) == 1 and aggregated_content[0].get(
                        "type") == "text":
                    content = aggregated_content[0]["text"]
```

(nit) can we put this is a helper function:

```
content = _aggregate_content(...) 
```
?